test-jest/cartesian/Brush.spec.tsx

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import { Brush, LineChart, Line, BarChart } from '../../src';
+import { mockMouseEvent } from '../helper/mockMouseEvent';
+
+describe('<Brush />', () => {
+  const data = [
+    { date: '2023-01-01', value: 10 },
+    { date: '2023-01-02', value: 20 },
+    { date: '2023-01-03', value: 10 },
+    { date: '2023-01-04', value: 30 },
+    { date: '2023-01-05', value: 50 },
+    { date: '2023-01-06', value: 10 },
+    { date: '2023-01-07', value: 30 },
+    { date: '2023-01-08', value: 20 },
+    { date: '2023-01-09', value: 10 },
+    { date: '2023-01-10', value: 70 },
+    { date: '2023-01-11', value: 40 },
+    { date: '2023-01-12', value: 20 },
+    { date: '2023-01-13', value: 10 },
+    { date: '2023-01-14', value: 10 },
+  ];
+
+  // wrap brush with a bar chart to make brush traveler work
+  const barChartWithBrush = (
+    <BarChart
+      width={500}
+      height={100}
+      data={data}
+      margin={{
+        right: 100,
+        left: 100,
+      }}
+    >
+      <Brush dataKey="date" height={90} stroke="#8884d8" />
+    </BarChart>
+  );
+
+  test('Render 2 travellers and 1 slide in simple Brush', () => {
+    const { container } = render(<Brush x={100} y={50} width={400} height={40} data={data} />);
+    expect(container.querySelectorAll('.recharts-brush-traveller')).toHaveLength(2);
+    expect(container.querySelectorAll('.recharts-brush-slide')).toHaveLength(1);
+  });
+
+  test("Don't render any travellers or slide when data is empty in simple Brush", () => {
+    const { container } = render(<Brush x={100} y={50} width={400} height={40} data={[]} />);
+    expect(container.querySelectorAll('.recharts-brush-traveller')).toHaveLength(0);
+    expect(container.querySelectorAll('.recharts-brush-slide')).toHaveLength(0);
+  });
+
+  test('Render panorama when specified LineChart as child', () => {
+    const { container } = render(
+      <Brush x={100} y={50} width={400} height={40} data={data}>
+        <LineChart>
+          <Line />
+        </LineChart>
+      </Brush>,
+    );
+
+    expect(container.querySelectorAll('.recharts-line')).toHaveLength(1);
+  });
+
+  test('mouse enter and mouse leave on traveller will set isTextActive true', () => {
+    const { container } = render(barChartWithBrush);
+
+    const brushSlide = container.querySelector('.recharts-brush-slide');
+
+    const mouseDownEvent = mockMouseEvent('mousedown', brushSlide!, { pageX: 0, pageY: 0 });
```

Comment:
The test title says MouseEnter and MouseLeave, but in the test we use mouseDown and mouseUp. 

While this test adds value it is not equivalent to the previous test. 

My suggestion: Add this tests as a new one, and also add the existing test with MouseEnter and MouseLeave.

Reply:
+1